connect_transformations/currency_conversion/mixins.py

```diff
@@ -30,38 +61,27 @@ async def currency_conversion(
         currency = trfn_settings['from']['currency']
         currency_to = trfn_settings['to']['currency']
 
+        async with self._current_exchange_rate_lock:
```

Comment:
This block should be moved to the `_get_current_exchange_rate` to avoid locking when the value is already available.